### DISCUSSION TITLE
"handleError" is not exported by "src/hooks.client.ts"

### DISCUSSION THREAD

**Discussion by orangecoloured** [sveltejs/kit#12737] [Q&A] [answered]
Recently I started to get this message when building my app.
```
.svelte-kit/generated/client-optimized/app.js (29:27): "handleError" is not exported by "src/hooks.client.ts", imported by ".svelte-kit/generated/client-optimized/app.js".
```
Can this be fixed? Seems like an internal `svelte` issue?

```
"@sveltejs/kit": "^2.6.1",
"vite": "^5.4.8"
```

**Comment by alexander-vidoni** [comment, marked as answer]
To remove this you have to create three files.

    /src/hooks.client.js
    /src/hooks.server.js
    /src/hooks.js

Then copy this code to _hooks.client.js_:

    ` /** @type {import('@sveltejs/kit').HandleServerError} */
    export async function handleError({ error, event, status, message }) {	
	return {};
    }`

Then copy this code to _hooks.js_

    ` /** @type {import('@sveltejs/kit').Reroute} */ 
    export function reroute({ url }) {}`

_hooks.server.js_ can be left empty, but it is also an entrypoint for some use cases, so create it now - some times you will need it. :-)

Both of these codes do currently nothing, but can be filled some day with logic when needed.

Further information can be found on the [Svelte Documentation | Hooks](https://kit.svelte.dev/docs/hooks)

**Reply by orangecoloured** [reply]
Perfect! Error is gone. 👌

**Reply by alexander-vidoni** [reply]
I am always happy to help!
